DNN Platform/Library/Entities/Portals/PortalAliasController.cs

```diff
@@ -1,390 +1,357 @@
+                // the startswith behaviour is preserved here to support those non-specific uses
+                var controller = new PortalAliasController();
+                var portalAliases = controller.GetPortalAliasesInternal();
+                var portalAliasCollection = portalAliases.OrderByDescending(k => k.Key.Length);
+
+                foreach (var currentAlias in portalAliasCollection)
+                {
+                    // check if the alias key starts with the portal alias value passed in - we use
+                    // StartsWith because child portals are redirected to the parent portal domain name
+                    // eg. child = 'www.domain.com/child' and parent is 'www.domain.com'
+                    // this allows the parent domain name to resolve to the child alias ( the tabid still identifies the child portalid )
+                    string httpAlias = currentAlias.Value.HTTPAlias.ToLowerInvariant();
+                    if (httpAlias.StartsWith(portalAlias.ToLowerInvariant()) && currentAlias.Value.PortalID == portalId)
+                    {
+                        retValue = currentAlias.Value.HTTPAlias;
+                        break;
+                    }
+
+                    httpAlias = httpAlias.StartsWith("www.") ? httpAlias.Replace("www.", string.Empty) : string.Concat("www.", httpAlias);
+                    if (httpAlias.StartsWith(portalAlias.ToLowerInvariant()) && currentAlias.Value.PortalID == portalId)
+                    {
+                        retValue = currentAlias.Value.HTTPAlias;
+                        break;
+                    }
+                }
+            }
+
+            return retValue;
+        }
+
+        /// <inheritdoc/>
+        string IPortalAliasService.GetPortalAliasByTab(int tabId, string portalAlias)
+        {
+            string retValue = Null.NullString;
+            int intPortalId = -2;
+
+            // get the tab
+            TabInfo tab = TabController.Instance.GetTab(tabId, Null.NullInteger);
+            if (tab != null)
+            {
+                // ignore deleted tabs
+                if (!tab.IsDeleted)
+                {
+                    intPortalId = tab.PortalID;
+                }
+            }
+
+            switch (intPortalId)
+            {
+                case -2: // tab does not exist
+                    break;
+                case -1: // host tab
+                    // host tabs are not verified to determine if they belong to the portal alias
+                    retValue = portalAlias;
+                    break;
+                default: // portal tab
+                    retValue = GetPortalAliasByPortal(intPortalId, portalAlias);
+                    break;
+            }
+
+            return retValue;
+        }
+
+        /// <inheritdoc />
+        bool IPortalAliasService.ValidateAlias(string portalAlias, bool ischild)

```

Comment:
While we're creating something new, can we fix the casing of this parameter?
```suggestion
        bool IPortalAliasService.ValidateAlias(string portalAlias, bool isChild)
```